daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -452,9 +454,30 @@ object NodeInfo extends Enum {
       }
     }
 
+    trait PrimThrowsIllegalArgument {
+      def fromString(s: String): DataValuePrimitive
+      def fromXMLString(s: String): DataValuePrimitive = {
+        try {
+          fromString(s)
+        } catch {
+          case iae: IllegalArgumentException =>
+            throw new InvalidPrimitiveDataException("%s is not a valid %s".format(iae.getMessage, this))
```

Comment:
Thoughts on using just ``s`` rather than ``iae.getMessage`` or ``nfe.getMessage`` and ignore whatever is in the exception? I'm concerened that the exception string isn't always going to make sense with "is not a valid %s". For example, it looks like we just use scala's ``String.toByte`` to for string to byte conversion. If a string is out of range for a byte, it results in the exception message
```
java.lang.NumberFormatException: Value out of range. Value:"256" Radix:10"
```
So this error message would become
```
Value out of range. Value:"256" Radix:10 is not a valid byte
```
which i find a little confusing. It's maybe nice that we get "out of range" information, but the radix info is just confusing and it doesn't fit well with the "is not a valid byte".

I'd also reccomend ``this.globalQName`` for the type, instead of just ``this``. That includes the XSD namespaces so makes it clear that the restriction is based on the XSD type system.

Reply:
I always like to just construct nested exceptions by passing the inner exception as a cause to the enclosing exception, and not constructing strings at all. 

There's an idiom for deriving exception classes so that you have both string and throwable constructors for them.  We don't do this for our exception classes, but we should so that you have string and cause constructors for them. Like this:

`class MyException protected (m: String, cause: Throwable)
extends Exception(m, cause) { // or extend an exception base class, 2-arg constructor
   def this(message: String) = this(message, null)
   def this(cause: Throwable) = this(null, cause)
  ....
}`

One can even omit the def this (m: String) constructor if the real usage pattern only involves nesting an exception.

Reply:
Also, since you don't know if an exception has a string message or a cause object, we have Misc.getSomeMessage(e) which returns a Some[String], i.e., you never get null back. Centralizes the logic of if you don't have a message, then check if there is a cause, and does it have a message or a cause, etc. etc.

Reply:
I think I just learned that you don't need the def this(..) in the code fragment above. That's inherited from the base.